native/src/main/java/io/xlibb/selenium/SeleniumWebDriver.java

```diff
@@ -0,0 +1,418 @@
+// Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package io.xlibb.selenium;
+
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BString;
+import io.xlibb.selenium.utils.Utils;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Dimension;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.firefox.FirefoxOptions;
+
+import java.util.List;
+import java.util.Set;
+
+public class SeleniumWebDriver {
+
+    public static final String DRIVER_OBJECT = "nativeDriverObject";
+    public static final String WEB_ELEMENT_OBJECT_TYPE = "WebElement";
+
+    public static Object openChrome(BObject webDriver, BMap<BString, Object> options) {
+        boolean headless = options.getBooleanValue(StringUtils.fromString("headlessMode"));
+        String url = options.getStringValue(StringUtils.fromString("url")).toString();
+        boolean incognito = options.getBooleanValue(StringUtils.fromString("incognitoMode"));
+        String[] additionalArguments = options.getArrayValue(StringUtils.fromString("additionalArguments"))
+                .getStringArray();
+        if (Utils.getDriverNObject(webDriver) == null) {
+            try {
+                ChromeOptions chromeOptions = new ChromeOptions();
+                if (headless) {
+                    chromeOptions.addArguments("--headless");
+                }
+                if (incognito) {
+                    chromeOptions.addArguments("--incognito");
+                }
+                for (String arg: additionalArguments) {
+                    chromeOptions.addArguments(arg);
+                }
+                webDriver.addNativeData(DRIVER_OBJECT, new ChromeDriver(chromeOptions));
+                Utils.getDriverNObject(webDriver).get(url);
```

Comment:
This will always return an error as `Utils.getDriverNObject(webDriver)` is already `null`. Btw could you explain the purpose of this line?